### DISCUSSION TITLE
Is there  source code for ekuiper-manager?

### DISCUSSION THREAD

**Discussion by iceteas** [lf-edge/ekuiper#1710] [General]
Is there  source code for ekuiper-manager?

**Comment by ngjaying** [comment]
No, eKuiper manager is not open sourced

**Reply by martinkbrown** [reply]
Thanks @ngjaying . Is there a way to request custom work to be done for ekuiper-manager? Or, if I would like to request and pay for some specific features to be added to ekuiper-manager to help meet some of my business objectives, is there a way for me to request that and fund that work? 

Thanks.

**Reply by ankur-paan** [reply]
We are building an open source eKuiper manager at [https://github.com/ankur-paan/ekuiper-manager](https://github.com/ankur-paan/ekuiper-manager). See if that helps. Also we are maintaining a live playground where you can interact with your own ekuiper instances using a browser at [https://ankur-paan.github.io/ekuiper-manager/](https://ankur-paan.github.io/ekuiper-manager/)

**Comment by mm7979** [comment]
Hi Martin. I work for EMQ and would be happy to discuss the enhancements to ekuiper-manager you are looking for. What is the best way to contact you?

**Reply by martinkbrown** [reply]
Hi @mm7979 , please email me at xmqxt1wq@duck.com . My email address might look weird, but it's real! Thanks.

**Reply by mm7979** [reply]
Martin, just sent you a second email regarding possible enhancements to eKuiper-manager. If you didn't see it perhaps it hit the junk folder. Ready to discuss, just waiting for your reply

**Reply by martinkbrown** [reply]
Ah, thanks. I got it! I'll respond over email.

**Comment by canob** [comment]
Is this the correct place to report an issue/bug on ekuiper-manager, or where I can report it?

**Reply by ngjaying** [reply]
@canob eKuiper manager is not maintained now. As far as I know, EMQ company is working on a new product which can also manage eKuiper. You can report the problem here, may be that could be fixed in that product.

**Reply by canob** [reply]
Ok, let's try if somebody from eKuiper Manager develop team wants to solve this problem, or maybe have a workaround for this, because it is really annoying.
I have the best of the user experiences with eKuiper Manager, the UI is amazing and really simple, but every time that I click on any rule, to see the content, I get this error in the UI, twice:
`invalid node id undefined, must be an uuid`
![image](https://github.com/lf-edge/ekuiper/assets/30271927/1675422f-6a6c-4ddd-a8cc-e25cdc43c093)
![image](https://github.com/lf-edge/ekuiper/assets/30271927/33e02f03-fa9c-4332-bdc8-1e574224f35e)
I can close both error messages and continue working, but is really annoying that every time I review a rule I have that two error messages, :(
Anyone with the same problem? Any workaround?
Thanks for your help.

**Reply by canob** [reply]
Sadly, I can confirm that this is still happening with the new version 1.9.5 of eKuiper Manager, :(

**Reply by ngjaying** [reply]
There is a 1.9.5-plus version. That should fix this problem

**Reply by canob** [reply]
Excellent!!! Thanks a lot!

**Comment by icoco** [comment]
possible let it open source ？

**Reply by ankur-paan** [reply]
We are building an open source eKuiper manager at https://github.com/ankur-paan/ekuiper-manager. See if that helps. Also we are maintaining a live playground where you can interact with your own ekuiper instances using a browser at https://ankur-paan.github.io/ekuiper-manager/

**Reply by canob** [reply]
Excellent news!
